### DISCUSSION TITLE
Generate Schema for getters without corresponding fields

### DISCUSSION THREAD

**Discussion by nickkhine** [victools/jsonschema-generator#380] [Q&A] [answered]
Hi is there anyway that I can generate the schema for getters that have no corresponding field easily? Tried a couple of the Options but having trouble to get this to work. 

So for example if I have the class: 
```java
public class TestGetter {
    private String fieldThatWillGenerate;

    public TestGetter() {
    }

    public String getFieldThatWillNotGenerate() {
        return "";
    }

    public String getFieldThatWillGenerate() {
        return this.fieldThatWillGenerate;
    }
}
```

and the code

```java
    @Test
    public void testGetterGeneration()
        throws JsonProcessingException {
        final ObjectMapper mapper = new ObjectMapper();
        final JacksonModule module = new JacksonModule();
        final SchemaGeneratorConfigBuilder configBuilder = new SchemaGeneratorConfigBuilder(mapper,
            SchemaVersion.DRAFT_2020_12,OptionPreset.PLAIN_JSON)
            .with(module);
        final SchemaGeneratorConfig config = configBuilder.build();
        final SchemaGenerator generator = new SchemaGenerator(config);
        final String actualJson = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(generator.generateSchema(TestGetter.class));
        System.out.println(actualJson);
    }
```

then the output would be 

```json
{
  "$schema" : "https://json-schema.org/draft/2020-12/schema",
  "type" : "object",
  "properties" : {
    "fieldThatWillGenerate" : {
      "type" : "string"
    }
  }
}
```

but I would like it to be 
```json
{
  "$schema" : "https://json-schema.org/draft/2020-12/schema",
  "type" : "object",
  "properties" : {
    "fieldThatWillGenerate" : {
      "type" : "string"
    },
    "fieldThatWillNotGenerate" : {
      "type" : "string"
    }
  }
}
```

so that it includes the suffix of the getter as the fieldname or the name using `@JsonProperty` if it is provided

**Comment by CarstenWickner** [comment, marked as answer]
HI @nickkhine,

I suggest you have a look at the documentation:
https://victools.github.io/jsonschema-generator/#generator-options

Since you're using the `OptionPreset.PLAIN_JSON`, you're missing the `Option.NONSTATIC_NONVOID_NONGETTER_METHODS` by default, which you should add explicitly in order for the `getFieldThatWillNotGenerate()` method to be considered.

Regarding the naming, `Option.FIELDS_DERIVED_FROM_ARGUMENTFREE_METHODS` would take care of your example.
The `JacksonModule` would use the `@JsonProperty` annotation's value instead if it is present.

However, this may include methods you don't want to be reflected in your JSON Schema (such as `toString()`).
Therefore you could either use the `JacksonOption.INCLUDE_ONLY_JSONPROPERTY_ANNOTATED_METHODS` on the `JacksonModule` and explicitly annotate each of the desired methods with `@JsonProperty` or you write your own configuration to ignore methods you don't want as per:
https://victools.github.io/jsonschema-generator/#omitting-ignoring-certain-fields-methods
e.g.
```java
configBuilder.forMethods()
    .withIgnoreCheck(method -> !method.getDeclaredName().startsWith("get"));
```

**Reply by nickkhine** [reply]
Ah yes, this combination seems to work. Thanks for the quick response!
